Web Inspector: (REGRESSION 297671@main) Oversized font-sizes in Audits and Graphics tabs

rcaliman-apple opened the pull request:
#### 8f765ec201581048d2116c0ed3720dad069242b6
<pre>
Web Inspector: (REGRESSION 297671@main) Oversized font-sizes in Audits and Graphics tabs
<a href="https://bugs.webkit.org/show_bug.cgi?id=297333">https://bugs.webkit.org/show_bug.cgi?id=297333</a>
<a href="https://rdar.apple.com/158226427">rdar://158226427</a>

Reviewed by BJ Burg.

Some User Agent styles for headings nested in `&lt;section&gt;` elements were removed in <a href="https://commits.webkit.org/297671@main">https://commits.webkit.org/297671@main</a>

The removed styles basically adjusted the heading styles down for each nesting level.
For example: `section h1` looks like `h2`, `section section h1` looks like `h3`.

This patch replicates the removed styles for the DOM structure in the Audits and Graphics tabs.

Drive-by: replace descendant element selectors with explicit direct descendant element selectors
to match Web Inspector style, clarify the DOM structure in CSS, and avoid unintentional styles.

* Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
(.content-view-container &gt; .content-view.audit-test-case.manager-editing &gt; header &gt; .information &gt; h1 &gt; img):
(.content-view.audit-test-case &gt; header &gt; .information &gt; h1):
(section &gt; .content-view.audit-test-case &gt; header &gt; .information &gt; h1):
(.content-view.audit-test-case &gt; header &gt; .information &gt; h1 &gt; img):
(.content-view.audit-test-case.manager-editing.disabled:not(.editable) &gt; header &gt; .information &gt; h1 &gt; img):
(.content-view.audit-test-case &gt; section &gt; :is(.dom-nodes, .errors) &gt; h1):
(section &gt; .content-view.audit-test-case &gt; section &gt; :is(.dom-nodes, .errors) &gt; h1):
(@media (prefers-color-scheme: dark) .content-view.audit-test-case.manager-editing &gt; header &gt; .information &gt; h1 &gt; img):
(.content-view-container &gt; .content-view.audit-test-case.manager-editing &gt; header h1 &gt; img): Deleted.
(.content-view.audit-test-case &gt; header h1): Deleted.
(.content-view.audit-test-case &gt; header h1 &gt; img): Deleted.
(.content-view.audit-test-case.manager-editing.disabled:not(.editable) &gt; header h1 &gt; img): Deleted.
(.content-view.audit-test-case &gt; section h1): Deleted.
(@media (prefers-color-scheme: dark) .content-view.audit-test-case.manager-editing &gt; header h1 &gt; img): Deleted.
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view-container &gt; .content-view.audit-test &gt; header &gt; .information &gt; h1):
(.content-view-container &gt; .content-view.audit-test &gt; header h1): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GraphicsOverviewContentView.css:
(.content-view.graphics-overview &gt; section &gt; .header &gt; h1):

Canonical link: <a href="https://commits.webkit.org/298633@main">https://commits.webkit.org/298633@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de18ad03fb213ba240753b56a30a99f929a575b6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/116117 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/35778 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/122173 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/66672 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/36472 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/44366 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/122173 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/66672 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/119065 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/36472 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/122173 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/36472 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/65855 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/36472 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125323 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/43011 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/44366 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/125323 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/43376 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125323 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24612 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/26322 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38957 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/42898 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/48490 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/42365 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/45700 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/44069 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/425b840c7cdafac429d543d78da17b572cefe168. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49335

rcaliman-apple commented:
# Fix screenshots

| Safari 26.0 | WebKit tip-of-tree + patch |
|---|---|
|<img width="1112" height="762" alt="audits-group-safari" src="https://github.com/user-attachments/assets/9907700a-464d-477c-9c53-5bd8cac27d74" />|<img width="1112" height="762" alt="audits-group-webkit" src="https://github.com/user-attachments/assets/3b158b82-44ff-441d-9287-5473209b6ec4" />|
|<img width="1112" height="762" alt="audits-group-edit-safari" src="https://github.com/user-attachments/assets/9ea03959-2471-41d3-86a3-6b18425e6ee1" />|<img width="1112" height="762" alt="audits-group-edit-webkit" src="https://github.com/user-attachments/assets/9bed0972-8392-4c58-9cac-30ae0654f21e" />|
|<img width="1112" height="762" alt="audits-test-safari" src="https://github.com/user-attachments/assets/10b40f5d-f78c-4241-bb6f-f9e3cda50c5f" />|<img width="1112" height="762" alt="audits-test-webkit" src="https://github.com/user-attachments/assets/17a84703-ec35-4933-aac1-ab0fdd467930" />|
|<img width="1112" height="762" alt="audits-test-edit-safari" src="https://github.com/user-attachments/assets/bb3f3c0c-8231-4d37-92f7-e36b27896170" />|<img width="1112" height="762" alt="audits-test-edit-webkit" src="https://github.com/user-attachments/assets/3f5dd61d-afd6-48a0-94a3-1beb4cf285fa" />|

burg commented:
Commit nit: 'headint'

rcaliman-apple committed:
Web Inspector: (REGRESSION 297671@main) Oversized font-sizes in Audits and Graphics tabs
https://bugs.webkit.org/show_bug.cgi?id=297333
rdar://158226427

Reviewed by BJ Burg.

Some User Agent styles for headings nested in `<section>` elements were removed in https://commits.webkit.org/297671@main

The removed styles basically adjusted the heading styles down for each nesting level.
For example: `section h1` looks like `h2`, `section section h1` looks like `h3`.

This patch replicates the removed styles for the DOM structure in the Audits and Graphics tabs.

Drive-by: replace descendant element selectors with explicit direct descendant element selectors
to match Web Inspector style, clarify the DOM structure in CSS, and avoid unintentional styles.

* Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
(.content-view-container > .content-view.audit-test-case.manager-editing > header > .information > h1 > img):
(.content-view.audit-test-case > header > .information > h1):
(section > .content-view.audit-test-case > header > .information > h1):
(.content-view.audit-test-case > header > .information > h1 > img):
(.content-view.audit-test-case.manager-editing.disabled:not(.editable) > header > .information > h1 > img):
(.content-view.audit-test-case > section > :is(.dom-nodes, .errors) > h1):
(section > .content-view.audit-test-case > section > :is(.dom-nodes, .errors) > h1):
(@media (prefers-color-scheme: dark) .content-view.audit-test-case.manager-editing > header > .information > h1 > img):
(.content-view-container > .content-view.audit-test-case.manager-editing > header h1 > img): Deleted.
(.content-view.audit-test-case > header h1): Deleted.
(.content-view.audit-test-case > header h1 > img): Deleted.
(.content-view.audit-test-case.manager-editing.disabled:not(.editable) > header h1 > img): Deleted.
(.content-view.audit-test-case > section h1): Deleted.
(@media (prefers-color-scheme: dark) .content-view.audit-test-case.manager-editing > header h1 > img): Deleted.
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view-container > .content-view.audit-test > header > .information > h1):
(.content-view-container > .content-view.audit-test > header h1): Deleted.
* Source/WebInspectorUI/UserInterface/Views/GraphicsOverviewContentView.css:
(.content-view.graphics-overview > section > .header > h1):

Canonical link: https://commits.webkit.org/298633@main
